src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -127,6 +137,73 @@ public void testUseConnectionCtx() throws SQLException {
     }
   }
 
+  /**
+   * This tests the ability to have quotes inside a schema or database. This fixes a bug where
+   * double-quoted function arguments like schemas, databases, etc were returning empty resultsets.
+   *
+   * @throws SQLException
+   */
+  @Test
+  public void testDoubleQuotedDatabaseAndSchema() throws SQLException {
+    try (Connection con = getConnection()) {
+      Statement statement = con.createStatement();
+      String database = con.getCatalog();
+      // To query the schema and table, we can use a normal java escaped quote. Wildcards are also
+      // escaped here
+      String querySchema = "TEST\\_SCHEMA\\_\"WITH\\_QUOTES\"";
+      String queryTable = "TESTTABLE\\_\"WITH\\_QUOTES\"";
+      // Create the schema and table. With SQL commands, double quotes must be escaped with another
+      // quote
+      statement.execute("create or replace schema \"TEST_SCHEMA_\"\"WITH_QUOTES\"\"\"");
+      statement.execute(
+          "create or replace table \"TESTTABLE_\"\"WITH_QUOTES\"\"\" (AMOUNT number, \"COL_\"\"QUOTED\"\"\" string)");
+      DatabaseMetaData metaData = con.getMetaData();
+      ResultSet rs = metaData.getTables(database, querySchema, queryTable, null);
+      // Assert 1 row returned for the testtable_"with_quotes"
+      assertEquals(1, getSizeOfResultSet(rs));
+      rs = metaData.getColumns(database, querySchema, queryTable, null);
+      // Assert 2 rows returned for the 2 rows in testtable_"with_quotes"
+      assertEquals(2, getSizeOfResultSet(rs));
+      rs = metaData.getColumns(database, querySchema, queryTable, "COL\\_\"QUOTED\"");
+      // Assert 1 row returned for the column col_"quoted"
+      assertEquals(1, getSizeOfResultSet(rs));
+      rs = metaData.getSchemas(database, querySchema);
+      // Assert 1 row returned for the schema test_schema_"with_quotes"
+      assertEquals(1, getSizeOfResultSet(rs));
+      rs.close();
+      statement.close();
+    }
+  }
+
+  /**
+   * This tests that wildcards can be used for the schema name for getProcedureColumns().
+   * Previously, only empty resultsets were returned.
+   *
+   * @throws SQLException
+   */
+  @Test
+  public void testGetProcedureColumnsWildcards() throws SQLException {
```

Comment:
Test that we can use a wildcard in the schema name of the function to get results from 2 schemas that both match the wildcard: SCH1 and SCH2